R/g.inspectfile.R

```diff
@@ -22,83 +22,69 @@ g.inspectfile = function(datafile, desiredtz = "", params_rawdata = c(),
-  # for convenience the monitor codes (mon):
-  # 0 - ad-hoc file (currently only .csv format)
-  # 1 - GENEA (non-commercial, DEPRECATED); 2 - GENEActiv
-  # 3 - Actigraph; 4 - Axivity (AX3, AX6)
-  # 5 - Movisense; 6 - Verisense
-  # data formats:
-  # 0 = ad-hoc .csv
-  # 1 - bin; 2 - csv; 3 - wav (DEPRECATED)
-  # 4 - cwa; 5 - movisens
-  # 6 - gt3x
-  
-  
+
   # note that if the file is an RData file then this function will not be called
   # the output of this function for the original datafile is stored inside the RData file in the form of object I
   getbrand = function(filename = c(), datafile = c()) {
     sf = c(); isitageneactive = c(); mon = c(); dformat = c() #generating empty variables
-    tmp1 = unlist(strsplit(filename,"[.]cs"))
-    tmp2 = unlist(strsplit(filename,"[.]b"))
-    tmp3 = unlist(strsplit(filename,"[.]w"))
-    tmp4 = unlist(strsplit(filename,"[.]r"))
-    tmp5 = unlist(strsplit(tolower(filename),"[.]cw")) # to lower to make this insensitive to case
-    tmp6 = unlist(strsplit(filename,"[.]gt3"))
-    tmp7 = unlist(strsplit(filename,"[.]GT3"))
-    if (tmp1[length(tmp1)] == "v" | tmp1[length(tmp1)] == "v.gz") { #this is a csv file
-      dformat = 2 #2 = csv
-      testcsv = read.csv(datafile, nrow = 10, skip = 10)
-      testcsvtopline = read.csv(datafile, nrow = 2,skip = 1)
-      if (ncol(testcsv) == 2 & ncol(testcsvtopline) < 4) { #it is a geneactivefile
-        mon = 2
-      } else if (ncol(testcsv) >= 3 & ncol(testcsvtopline) < 4) {	#it is an actigraph file
-        mon = 3
-      } else if (ncol(testcsv) >= 4 & ncol(testcsvtopline) >= 4) { # it is an AX3 file
-        mon = 4
-      }
-    } else if (tmp2[length(tmp2)] == "in") { #this is a bin file
-      dformat = 1 #1 = binary
-    } else if (tmp5[length(tmp5)] == "a") { #this is a cwa file
-      dformat = 4 #4 = cwa
-      mon = 4 # Axivity
-    } else if (tmp6[length(tmp6)] == "x") { #this is a gt3x file
-      dformat = 6 #6 = gt3x
-      mon = 3 # actigraph
-    } else if (tmp7[length(tmp7)] == "X") { #this is a gt3x file from Centerpoint
-      if (file.access(datafile, 2) == 0) { # test for write access to file
-        # rename file to be lower case gt3x extension
-        file.rename(from = datafile, to = gsub(pattern = ".GT3X", replacement = ".gt3x", x = datafile))
-        datafile = gsub(pattern = ".GT3X", replacement = ".gt3x", x = datafile)
-        warning("\nWe have renamed the GT3X file to gt3x because GGIR dependency read.gt3x cannot handle uper case extention")
-        dformat = 6 #6 = gt3x
-        mon = 3 # actigraph
-      } else {
-        stop("\nGGIR wants to change the file extension from GT3X to gt3x, but it does not seem to have write permission to the file.")
-      }
-    }
-    is.mv = ismovisens(datafile)
-    if (is.mv == TRUE) {
-      dformat = 1
+
+    extention = unlist(strsplit(filename,"[.]"))[2]
+
+    switch (extention,
+      "bin" = { dformat = FORMAT$BIN },
+      "cwa" = ,
```

Comment:
`"cwa" = ,`

should this be

```
"cwa" = { mon = MONITOR$AXIVITY
                dformat = FORMAT$CWA
```

Reply:
Update: I have now addressed this in PR #884

Reply:
We can add this like you suggested, but I think it's better not to. with `switch`, if we provide a name but no element to evaluate, then whenever that name matches, we fall through to the next element that isn't missing.

this is from the help for `switch`:

> If EXPR evaluates to a character string then that string is matched (exactly) to the names of the elements in .... If there is a match then that element is evaluated unless it is missing, in which case the next non-missing element is evaluated, so for example switch("cc", a = 1, cc =, cd =, d = 2) evaluates to 2.`

 So I think it's better to leave `"cwa" = ,` immediately followed by `"CWA" = { mon = MONITOR$AXIVITY<...>` This way it's clear that "cwa" and "CWA" should have an identical effect, and we don't accidentally end up with a different result for `cwa` and `CWA`.

But, either way works, so please let me know what you prefer!

Reply:
Ok, I didn't knew that. No problem, then let's keep it this way.